tests/vmi_cloudinit_test.go

```diff
@@ -486,18 +421,18 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 				vmi := libvmifact.NewCirros(
 					libvmi.WithCloudInitConfigDrive(libcloudinit.WithConfigDriveEncodedNetworkData(testNetworkData)),
 				)
-				vmi = LaunchVMI(vmi)
+				vmi = libvmops.RunVMIAndExpectLaunch(vmi, startupTime)
 				vmi = libwait.WaitUntilVMIReady(vmi, console.LoginToCirros)
 
-				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceConfigDrive)
+				checkCloudInitIsoSize(vmi, cloudinit.DataSourceConfigDrive)
 				By("mouting cloudinit iso")
-				MountCloudInitConfigDrive(vmi)
+				tests.MountCloudInitFunc("config-2")(vmi)
 
 				By("checking cloudinit network-config")
-				CheckCloudInitFile(vmi, "openstack/latest/network_data.json", testNetworkData)
+				checkCloudInitFile(vmi, "openstack/latest/network_data.json", testNetworkData)
 
 			})
-			It("[test_id:3186]should have cloud-init network-config from k8s secret", func() {
+			It("[test_id:3186]should have cloud-init network-config from k8s secret", decorators.Conformance, func() {
```

Comment:
Must we add all 6 cases to Conformance? Each of them represents an inherent race condition (we wait for a VM to start, not knowing until when) which is a bad practive. Most of the cloudinit functionality should be covered by tests on the code converting vmi yaml to cloudinit disk. In many senses, we test here the cloudinit package of Fedora and CentOS rather than KubeVirt.

I'm not asking to delete these test, but let us keep Conformance lean and short (while representative). Can you choose just one of the tests? The others can keep running in less critical lanes.

Reply:
I don't agree with the race condition statement but let's leave it aside.

>Most of the cloudinit functionality should be covered by tests on the code converting vmi yaml to cloudinit disk. In many senses, we test here the cloudinit package of Fedora and CentOS rather than KubeVirt.

I agree but that is implementation detail we can change at any point of time. Nevertheless we interact with this tool and we need to have integration tests (not saying it needs to be e2e). 

>Can you choose just one of the tests?

This is hard for me to argue. Here I picked one test for each case (cloudInitNoCloud, cloudInitConfigDrive,  + secret). I had to include the network related because they are not part of the ones which exists but here I don't really see why we can't merge them. I wanted to ask around if anyone sees a reason to keep the separate at the basic ones that I think should be in Conformance. So hopefully we can have only 4 tests in the end (as part of Conf.)
Does this make sense?

@EdDev would you see problem with merging the network related tests?

Reply:
> I don't agree with the race condition statement but let's leave it aside.

We can leave it aside, but it is very important. Every time we start a VM, we do not know when it would become ready. We have a guess, we have a timeout, and we have a race condition between the starting VM and our timeout. This race condition brings about inconsistent results when we run the tests on different platforms and it slows us down considerably.

This is inherent to out eventually-consistent platform. We cannot run away from it. But we can make it less bad by starting fewer VMs in e2e tests, and certainly in the Conformance suite.

FWIW I think that we should merge tests. Or more precisely, use fixtures to create one VM, and then have multiple read-only tests working against it.

Reply:
>FWIW I think that we should merge tests. Or more precisely, use fixtures to create one VM, and then have multiple read-only tests working against it.

Would you require this in this PR or can we follow up on this? I would like to have first version fast and then iterate...

Reply:
> > FWIW I think that we should merge tests. Or more precisely, use fixtures to create one VM, and then have multiple read-only tests working against it.
> 
> Would you require this in this PR or can we follow up on this? I would like to have first version fast and then iterate...

This PR should stay small. But I would like it to keep Conformance VM startup count low. I wish you can choose only one test in this file; more tests can be added once they are made less expansive, e.g by reusing VMs.

Reply:
I believe it is on current possible minimum, we cannot use the common setup as both cloud options require different VM definition.

Reply:
> I believe it is on current possible minimum, we cannot use the common setup as both cloud options require different VM definition.

So no more tests would be ever added to Conformance.

We always have a tension between test speed and coverage. For Conformance we should strongly err towards making test quick.

Reply:
... in other words: please choose only one test to add to Conformance. The rest of them can stay in the regular sig-compute lane.

Reply:
> @EdDev would you see problem with merging the network related tests?

You can merge these into other tests if it fits.
We already use some cloud-init network functionality to define the network configuration on the VM.
So you could also check which ones are already covered and drop them from here as an additional optimization or alternative.

Reply:
> ... in other words: please choose only one test to add to Conformance. The rest of them can stay in the regular sig-compute lane.

As this is only relevant for the last commit, you could also consider the marking of conformance on the tests as a follow up separate PR. It will be quicker to take in the agree part and discuss this last thing separately.

Reply:
@dankenigsberg Now we have 3 cases included.
1. ConfigDrive
2. NoCloud with k8s secret (different flow than 1.)
3. stability of instance-id (This is important attribute)

Reply:
(I'm not a bit fan of the third one)